packages/calypso-e2e/src/lib/pages/plans-page.ts

```diff
@@ -97,31 +152,31 @@ export class PlansPage {
 	 * @throws If the expected tab name is not the active tab.
 	 */
 	async validateActiveNavigationTab( expectedTab: PlansPageTab ): Promise< void > {
-		await this.waitUntilLoaded();
-
-		// For mobile sized viewport, the currently selected tab name will be shown alongside the
-		// dropdown toggle button, so verify the expected tab name is shown there.
+		// For mobile sized viewport, the currently selected tab name
+		// is hidden behind a pseudo-dropdown.
+		// Therefore the valicdation will look for hidden element.
+		const currentSelectedLocator = this.page.locator(
+			selectors.activeNavigationTab( expectedTab )
+		);
 		if ( envVariables.VIEWPORT_NAME === 'mobile' ) {
-			await this.page.waitForSelector(
-				`${ selectors.mobileNavTabsToggle }:has-text("${ expectedTab }")`
-			);
+			await currentSelectedLocator.waitFor( { state: 'hidden' } );
 		} else {
-			await this.page.waitForSelector( selectors.activeNavigationTab( expectedTab ) );
+			await currentSelectedLocator.waitFor();
 		}
 	}
 
 	/**
 	 * Click a plan action button (on the plan cards on the "Plans" tab) based on expected plan name and button text.
 	 *
 	 * @param {object} param0 Object containing plan name and button text
-	 * @param {Plan} param0.plan Name of the plan (e.g. "Premium")
+	 * @param {LegacyPlans} param0.plan Name of the plan (e.g. "Premium")
 	 * @param {PlanActionButton} param0.buttonText Expected action button text (e.g. "Upgrade")
 	 */
 	async clickPlanActionButton( {
```

Comment:
I chose not to update the method name here:
- it adds a lot of unnecessary changes to the PR;
- if a developer tries to use this for the Pro plan, the lack of such option quickly becomes apparent.